packages/api/helper/gmailAPI.js

```diff
@@ -0,0 +1,22 @@
+const opn = require('open');
+const { google } = require('googleapis');
+const credentials = require('./credentials.json');
+
+function gmailStart() {
+  const SCOPES = ['https://www.googleapis.com/auth/gmail.modify'];
+  const oauth2Client = new google.auth.OAuth2(
+    credentials.installed.client_id,
+    credentials.installed.client_secret,
+    credentials.installed.redirect_uris[0],
+  );
+  google.options({ auth: oauth2Client });
+
+  const authorizeUrl = oauth2Client.generateAuthUrl({
+    access_type: 'offline',
+    scope: SCOPES,
+  });
+
+  opn(authorizeUrl, { wait: true, newInstance: false }).then((cp) => cp.unref());
```

Comment:
I'm not sure this will work at the API level when deployed. I believe it works on your local machine because the API is running there, so `open` opens a browser window. But on the server, it won't be able to open the user's browser window.

You may need to have the front end handle this part. Typically, the front end would launch the Google auth page, then in this case the `redirect_uri` might point at the backend.

Reply:
How do we test if it will work when deployed ? Is this something that has to wait for heroku testing ?